Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -9,13 +9,17 @@ import Foundation
 
 class HttpClient {
     let configuration: Configuration
-    internal var session: URLSession
+    internal let session: URLSession
 
     init(configuration: Configuration) {
         self.configuration = configuration
-        // shared instance has limitations but think we are not affected
-        // https://developer.apple.com/documentation/foundation/urlsession/1409000-shared
-        self.session = URLSession.shared
+        // an exclusive ephemeral session avoids contention (fewer request timeouts)
+        let sessionConfiguration = URLSessionConfiguration.ephemeral
+        // add some throughput
+        sessionConfiguration.httpMaximumConnectionsPerHost = 2
```

Comment:
[Default is 6](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/1407597-httpmaximumconnectionsperhost#discussion). Why reduce it to 2?

Reply:
Happy with 6 if you are!